src/java/org/apache/cassandra/utils/memory/EnsureOnHeap.java

```diff
@@ -32,6 +32,8 @@
 
 public abstract class EnsureOnHeap extends Transformation
 {
+    public static final EnsureOnHeap NOOP = new NoOp();
```

Comment:
It was needed for `AbstractMemtable.AbstractFlushCollection#iterator` to avoid the on-heap copying (a feature of DSE 6 I'd like to bring over). This currently doesn't work and leaves this change unused, but I do intend to solve it soon.